src/mcp/tools/dataconnect/generate_dataconnect_operation.ts

```diff
@@ -0,0 +1,42 @@
+import { z } from "zod";
+import { tool } from "../../tool.js";
+import { toContent } from "../../util.js";
+import { generateOperation } from "../../../gif/fdcExperience.js";
+import { pickService } from "../../../dataconnect/fileUtils.js";
+
+export const generate_dataconnect_operation = tool(
+  {
+    name: "generate_dataconnect_operation",
+    description:
+      "Generates a single Firebase Data Connect query or mutation based on the currently deployed schema and the provided prompt.",
+    inputSchema: z.object({
+      // Lifted guidance from : https://cloud.google.com/gemini/docs/discover/write-prompts
+      prompt: z
+        .string()
+        .describe(
+          "Write the prompt like you're talking to a person, describe the task you're trying to accomplish and give details that are specific to the users requst",
+        ),
+      serviceId: z
+        .string()
+        .nullish()
+        .describe(
+          "Optional: Uses the serviceId from the firebase.json file if nothing provided. The serviceId of the deployed Firebase resource.",
+        ),
```

Comment:
This is effectively the generate_connector in console. It would use the deployed schema in the backend~

It's a great starting point, but I think we would want one with local schema.

LGTM. Unblock you~

Reply:
Yes, I agree. Not sure if GiF can handle local schema. Maybe we can fix in future update if available?